packages/cli/test/unit/util/get-flags-specification.test.ts

```diff
@@ -0,0 +1,113 @@
+import { describe, it, expect } from 'vitest';
+import { CommandOption } from '../../../src/commands/help';
+import { getFlagsSpecification } from '../../../src/util/get-flags-specification';
+
+describe('getFlagsSpecification', () => {
+  it('success - return empty object when given no options', () => {
+    expect(getFlagsSpecification([])).toMatchObject({});
+  });
+
+  it('success - handles boolean options', () => {
+    const options = [
+      { name: 'an-option', type: Boolean, shorthand: null, deprecated: false },
+      {
+        name: 'another-option',
+        type: Boolean,
+        shorthand: null,
+        deprecated: false,
+      },
+    ];
+    expect(getFlagsSpecification(options)).toMatchObject({
+      '--an-option': Boolean,
+      '--another-option': Boolean,
+    });
+  });
+
+  it('success - handles string options', () => {
+    const options = [
+      { name: 'an-option', type: String, shorthand: null, deprecated: false },
+      {
+        name: 'another-option',
+        type: String,
+        shorthand: null,
+        deprecated: false,
+      },
+    ];
+    expect(getFlagsSpecification(options)).toMatchObject({
+      '--an-option': String,
+      '--another-option': String,
+    });
+  });
+
+  it('success - handles string array options', () => {
+    const options: CommandOption[] = [
+      { name: 'an-option', type: [String], shorthand: null, deprecated: false },
+      {
+        name: 'another-option',
+        type: [String],
+        shorthand: null,
+        deprecated: false,
+      },
+    ];
+    expect(getFlagsSpecification(options)).toMatchObject({
+      '--an-option': [String],
+      '--another-option': [String],
+    });
+  });
+
+  it('success - handles number options', () => {
+    const options = [
+      { name: 'an-option', type: Number, shorthand: null, deprecated: false },
+      {
+        name: 'another-option',
+        type: Number,
+        shorthand: null,
+        deprecated: false,
+      },
+    ];
+    expect(getFlagsSpecification(options)).toMatchObject({
+      '--an-option': Number,
+      '--another-option': Number,
+    });
+  });
```

Comment:
<!-- Suggestion comments propose improvements to the code. Mention specifically what should be changed and what the benefit would be.  -->
**suggestion (non-blocking):** ‏consider removing

I don't think these provide much value over the combo test below. Failure of the combo test will still be very clear about which type isn't working. Do you think we need these tests?

I don't mind keeping them. They're very fast.

Reply:
I chose to enumerate them specifically because I want to make sure that we continue to support these types in the event that we change argument parsers in the future.

I think the mixed test muddies this intent. It does cover an important case, but doesn't clearly say things about the types that individual tests do better.

Reply:
I don't understand how the individual tests say something better. Are you saying that having the separate test at all clarifies the intent to support that type?

Reply:
Yes, that was my intention.

Reply:
I don't agree that this overly conveys that over the combo test, but I don't mind keeping them all.